src/waitFor.js

```diff
@@ -1,16 +1,27 @@
 // @flow
 
+import React from 'react';
+import act from './act';
 import { throwRemovedFunctionError } from './helpers/errors';
 
 const DEFAULT_TIMEOUT = 4500;
 const DEFAULT_INTERVAL = 50;
 
+function checkReactVersionAtLeast(major: number, minor: number): boolean {
+  if (React.version === undefined) return false;
+  const [actualMajor, actualMinor] = React.version
+    .split('.')
+    .map((n) => parseInt(n, 10));
+
+  return actualMajor > major || (actualMajor == major && actualMinor >= minor);
```

Comment:
We don't do `==` here, it's dangerous in JS :D 
```suggestion
  return actualMajor > major || (actualMajor === major && actualMinor >= minor);
```